APIを用いたGithubリポジトリの一覧表示

Kate941-su opened the pull request:
# ステート管理
Riverpod

# エンティティクラスの作成
freezed

# API通信パッケージ
dio

Commit:
First Commit

Commit:
画面のレイアウトを実装

Commit:
レイアウト作成

Commit:
fvm設定ファイルをバージョン管理下に追加

Commit:
fvm_config.jsonのみバージョン管理下に配置

Commit:
fvm_config.jsonにバージョンを追記

Commit:
appbarのTextFieldをウィジェット化

Commit:
ListTileウィジェットの修正

HiroyukiTamura committed:
memo.txtを削除

Commit:
freezedを用いたファイルの自動生成を実装。dioを用いたAPIwosiyousiteデータをフェッチするコードを実装

Commit:
検索機能ようstateNotifyProviderを追加

Commit:
検索機能用Notifierの追加

Commit:
- 状態管理周りのコードを修正
- 全体のリファクタリング

Commit:
検索機能の追加

HiroyukiTamura reviewed (commented):
![next-badge](https://img.shields.io/badge/review-next-blueviolet)

`flutter pub run build_runner build`や`flutter pub run build_runner build -d`といったコマンドをターミナルから打つことがよくあると思います。

コマンドのテンプレートを作ってgitにpushすると、他の開発者とテンプレートを共有することができるので便利です。
IDEのRun/Edit Configurtionを開き、テンプレを作成・ファイルとして保存してみましょう。

<img width="1088" alt="スクリーンショット 2023-05-30 16 09 25" src="https://github.com/HiroyukiTamura/FlutterTraining/assets/26954245/94ad735e-bca6-48de-9f19-4f37341d0e1c">

HiroyukiTamura commented on lib/hole_view.dart:
```diff
@@ -0,0 +1,76 @@
+    String name = '';
+    // WidgetRef
+    // 取得したAPIデータの監視
+    final asyncValue = ref.watch(listProvider);
```
![good-badge](https://img.shields.io/badge/review-good-blightgreen.svg)

理想的なコードだと思います。
`riverpod`の`AsyncValue`、便利ですよね。freezedでも`when`はよく使いますし、switch文ではなくこのようにこのようにスッキリ書けるあたり、riverpod/freezedの便利さが実感できると思います。

HiroyukiTamura commented on lib/hole_view.dart:
```diff
@@ -0,0 +1,76 @@
+                // null非許容型としてキャストする
+                final elements = data![index];
+                // Nullチェックを行う
+                stargazersCount = elements.stargazers_count ?? 0;
```
![good-badge](https://img.shields.io/badge/review-good-blightgreen.svg)

`??`を活用していますね、とてもよい書き方だと思います。

HiroyukiTamura commented on lib/hole_view.dart:
```diff
@@ -0,0 +1,76 @@
+              },
+              separatorBuilder: (context, index) {
+                return const Divider(
+                  height: 0.5,
```
![imo-badge](https://img.shields.io/badge/review-imo-orange.svg)

とても些細ですが、dartでは1の位が0の小数であれば1の位を省略できます。(ex. `0.5` ->` .5`)

HiroyukiTamura commented on lib/repository_items.dart:
```diff
@@ -0,0 +1,18 @@
+  factory RepositoryItems({
+    String? name,
+    String? description,
+    int? stargazers_count,
```
![must-badge](https://img.shields.io/badge/review-must-red.svg)

変数はcamel caseで記述しましょう。
ここでは、確かにjsonのkeyが`stargazers_count`ですが、変数名は別名を定義することができます。
freezedにドキュメントに載っているので、是非書き直してみてください。

HiroyukiTamura reviewed (commented):
![good-badge](https://img.shields.io/badge/review-good-blightgreen.svg)
全体的にかなりよく書けていると思います。
freezedの`@Default`などのアノテーションや、各種providerなど、理解が進んでいることが伝わってきました。
この調子で手を動かながら理解していくのを今後も続けられたらいいですね。


![must-badge](https://img.shields.io/badge/review-must-red.svg)

AndroidStudioのDartAnalysisを開くと、freezedで自動生成されたファイルもanalysisの対象になってしまっています。
feezedのドキュメントに従って、analysisを編集してみてください。
そのうえで、analysisのワーニングを全て消してみましょう。


![memo-badge](https://img.shields.io/badge/review-memo-lightgrey)

レイアウトの組んでいるとき、意図したとおりにUIが組めず困る時ってありますよね。
AndroidのFlutter Inspectorを開いて、右上のボタンからレイアウトのガイドラインを表示すると、レイアウト構造をUI上に表示できます。困ったときに使ってみてください。
<img width="803" alt="スクリーンショット 2023-05-30 16 41 46" src="https://github.com/HiroyukiTamura/FlutterTraining/assets/26954245/00d15b3e-aea9-40d8-bce1-574d5a271be3">

Commit:
- Add run configration file(実行の設定ファイルを追加)
- remove Dart Analysis warning (Dart Analysisの警告を削除)
- fix freezed charactor of  member variables (freezedのメンバ変数の文字を変更)

Commit:
ビルドの設定ファイルの名前を変更
